### ISSUE TITLE
Save Changes doesn't exit the config menu

### ISSUE THREAD

**Issue by falseresync** [isXander/YetAnotherConfigLib#142]
I have:
1. Setup [a config with AutoGen](https://github.com/falseresync/wizcraft/blob/main/src/main/java/dev/falseresync/wizcraft/common/WizcraftConfig.java)
1. Setup [a ModMenu integration ](https://github.com/falseresync/wizcraft/blob/main/src/main/java/dev/falseresync/wizcraft/compat/modmenu/WizcraftModMenuCompat.java)

When I click `Save changes`, no matter if I am in Game or on the Main menu, it does save the config file, but it doesn't exit the config screen.
The only way to exit is to click Cancel, or to click Undo (which doesn't undo the changes because I have already clicked Save changes) and then click Done

**Comment by Ayydxn** [comment]
A workaround you can do for now and what I'm also doing in [my project](https://github.com/Ayydxn/Iridium) is just to close the screen yourself.

This is what I'm doing:
```
public Screen getHandle()
    {
        return YetAnotherConfigLib.createBuilder()
                .title(Text.translatable("iridium.options.gui_title"))
                .categories(this.optionCategories)
                .save(() ->
                {
                    this.client.options.write();
                    this.iridiumGameOptions.write();

                    // TODO: (Ayydan) Remove this when YACL has fixed the issue with the "Save Changes" button not working properly.
                    //  Refer to this GitHub issue on its repo for more details: https://github.com/isXander/YetAnotherConfigLib/issues/142
                    this.client.setScreen(this.parentScreen);
                })
                .build()
                .generateScreen(this.parentScreen);
    }
```

Hope this helps.

**Comment by falseresync** [comment]
Thanks again!

**Comment by falseresync** [comment]
Fixed in v3.3.2
